### ISSUE TITLE
Passthrough for error types that are already IntoResponse

### ISSUE THREAD

**Issue by valpackett** [Shyrogan/axum_thiserror#1]
This looks very convenient! :+1: 

One thing that seems missing is being able to reuse an error's own `IntoResponse` implementation when doing `#[from]`.
e.g. [JsonDeserializerRejection](https://docs.rs/axum-extra/0.9.3/axum_extra/extract/enum.JsonDeserializerRejection.html) is such an error.

**Comment by Shyrogan** [comment]
Indeed, I’ll look into that when I have free time, if you have time or want to implement it just make a PR and let me know :)
